File: iOS4bookExamples/p740p758XML/p740XML/MyPeopleParser.m


#import "MyPeopleParser.h"
#import "MyPersonParser.h"

@implementation MyPeopleParser
@synthesize people;


- (void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qualifiedName attributes:(NSDictionary *)attributeDict
{
    if ([elementName isEqualToString: @"people"])
        self.people = [NSMutableArray array];
    if ([elementName isEqualToString: @"person"]) {
        [self makeChild:[MyPersonParser class] elementName:elementName parser:parser];
    }
}

- (void) finishedChild: (NSString*) s {
    [people addObject: [(MyPersonParser*)self.child person]];
}


@end

File: iOS10bookExamples/bk2ch23p818xml/ch36p1070xml/MyPeopleParser.swift


import UIKit

class MyPeopleParser : MyXMLParserDelegate {
    
    var people = [Person]()
    
    func parser(_ parser: XMLParser, didStartElement elementName: String,
        namespaceURI: String?, qualifiedName qName: String?,
        attributes attributeDict: [String : String]) {
            if elementName == "person" {
                self.makeChild(MyPersonParser.self, elementName: elementName, parser: parser)
            }
    }
    override func finishedChild(_ s: String) {
        self.people.append((self.child as! MyPersonParser).person)
    }
    
}
